src/main/main.ts

```diff
@@ -125,13 +127,13 @@
     .whenReady()
     .then(async () => {
       createWindow();
-      registerIpcHandlers();
       initSocket();
-      watchDatabase();
-      spawnBalanceCheck();
-      watchElectrumTransactions();
-      watchLogs();
-      initStats();
+      startRPC();
+
+      setInterval(() => {
+        getRawSwapInfos();
+      }, 1000 * 20);
+
       return 0;
     })
     .catch((e) =>
```

Comment:
**Suggestion:** Consider using a more robust error handling strategy. Currently, if an error occurs during the execution of the 'whenReady' function, the error is simply logged and the execution continues. This might lead to unexpected behavior later in the execution. Consider throwing the error or handling it in a way that prevents further execution in case of a critical failure.
```suggestion
    try {
      await app.whenReady();
      createWindow();
      initSocket();
      startRPC();
      setInterval(() => {
        getRawSwapInfos();
      }, 1000 * 20);
    } catch (e) {
      logger.error(
        { error: e.toString() },
        'Failed to initialize application'
      );
      throw e;
    }
```